File: twitter_sentiment/tw_sentiment.jl
# import TextAnalysis
import JSON
import Twitter

twitter_config_file = open("$(homedir())/.twitter/twitter_sentiment1111.json")
tw_config = JSON.parse(twitter_config_file)

# println(tw_config)

Twitter.twitterauth(
            tw_config["consumer_key"],
            tw_config["consumer_secret"],
            tw_config["access_token"],
            tw_config["access_token_secret"])



# duke_tweets = Twitter.get_search_tweets(q = "Trump")
# tw = duke_tweets[1]

# println(tw)

# str = "To be or not to be..."
# sd = StringDocument(str)

# text(sd)

# language(sd)
# name(sd)
# author(sd)
# timestamp(sd)

File: twitter_sentiment/tw_sentiment.py
import tweepy
from textblob import TextBlob
import json
from pathlib import Path
from pprint import pprint

home = str(Path.home())
with open('%s/.twitter/twitter_sentiment1111.json' % home) as f:
    tw_config = json.load(f)


consumer_key = tw_config['consumer_key']
consumer_secret = tw_config['consumer_secret']
access_token = tw_config['access_token']
access_token_secret = tw_config['access_token_secret']

auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)

api = tweepy.API(auth)

public_tweets = api.search('Trump')

for tweet in public_tweets:
    print(tweet.text)
    analysis = TextBlob(tweet.text)
    print(analysis.sentiment)
    
# pprint(consumer_secret)



